powersimdata/input/transform_profile.py

```diff
@@ -1,111 +1,98 @@
 import copy
 
-from powersimdata.input.grid import Grid
 from powersimdata.input.input_data import InputData
+from powersimdata.network.usa_tamu.constants.zones import interconnect2id
 
 
 class TransformProfile(object):
-    """Transforms profile according to operations listed in change table."""
+    """Transform profile according to operations listed in change table."""
 
     def __init__(self, scenario_info, grid, ct):
-        """Constructor
+        """Constructor.
 
         :param dict scenario_info: scenario information.
         :param powersimdata.input.grid.Grid grid: a Grid object.
         :param dict ct: change table.
         """
         self._input_data = InputData()
         self.scenario_info = scenario_info
-        self.grid = copy.deepcopy(grid)
+
         self.ct = copy.deepcopy(ct)
+        self.grid = copy.deepcopy(grid)
+
         self.scale_keys = {
-            "wind": {"wind", "wind_offshore"},
-            "solar": {"solar"},
-            "hydro": {"hydro"},
-            "demand": {"demand"},
+            "wind": ["wind", "wind_offshore"],
+            "solar": ["solar"],
+            "hydro": ["hydro"],
+            "demand": ["demand"],
         }
-        self.n_new_plant = (
-            0 if "new_plant" not in self.ct.keys() else len(self.ct["new_plant"])
-        )
-        self.base_plant = Grid(self.grid.interconnect).plant
+        self.n_new_plant, self.n_new_clean_plant = self._get_number_of_new_plant()
+
+    def _get_number_of_new_plant(self):
+        """Return the total number of new plant and new clean plant.
+
+        :return: (*tuple*) -- first element is the total number of new plant and second
+            element is the total number of new clean plant (*hydro*, *solar*,
+            *onshore wind* and *offshore wind*).
+        """
+        n_plant = [0, 0]
+        if "new_plant" in self.ct.keys():
+            for p in self.ct["new_plant"]:
+                n_plant[0] += 1
+                if p["type"] in sum(self.scale_keys.values(), []):
+                    n_plant[1] += 1
+        return n_plant
 
     def _get_renewable_profile(self, resource):
-        """Returns the transformed grid.
+        """Return the transformed profile.
 
         :param str resource: *'hydro'*, *'solar'* or *'wind'*.
-        :return: (*pandas.DataFrame*) -- power output for generators of
-            specified type with plant identification number  as columns and
-            UTC timestamp as index.
+        :return: (*pandas.DataFrame*) -- power output for generators of specified type
+            with plant identification number as columns and UTC timestamp as indices.
         """
-        power_output = self._input_data.get_data(self.scenario_info, resource)
-        if not bool(self.ct):
-            return power_output
+        plant_id = (
+            self.grid.plant.iloc[: len(self.grid.plant) - self.n_new_plant]
```

Comment:
Can this line be simplified to just `self.grid.plant.iloc[:-self.n_new_plant]`?

Reply:
I tried but If I remember correctly, I got an error when `self.n_new_plant` equals 0 with `self.grid.plant.iloc[:-self.n_new_plant]`

Reply:
Aha, that makes sense.